docs/machine-learning/tutorials/text-classification-tf.md

```diff
@@ -0,0 +1,260 @@
+
+`IMDBPrediction` is the prediction class used after the model training. `IMDBPrediction` has a single `float` array (`Prediction`) and a `VectorType` attribute.
+
+### Adding context
+
+The [MLContext class](xref:Microsoft.ML.MLContext) is a starting point for all ML.NET operations. Initializing `mlContext` creates a new ML.NET environment that can be shared across the model creation workflow objects. It's similar, conceptually, to `DBContext` in Entity Framework.
+
+1. Replace the `Console.WriteLine("Hello World!")` line in the `Main` method with the following code to declare and initialize the mlContext variable:
+
+   [!code-csharp[CreateMLContext](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#CreateMLContext "Create the ML Context")]
+
+1. Add the following as the next line of code in the `Main()` method:
+
+   [!code-csharp[CallReuseAndTuneSentimentModel](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#CallReuseAndTuneSentimentModel)]
+
+### Create an intermediate features class
+
+1. Create a utility class called `IntermediateFeatures` to hold intermediate data that will be used by the CustomMapping Estimator in a later step. Add this after the `Main()` method:
+
+[!code-csharp[DeclareIntermediateFeatures](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#DeclareIntermediateFeatures)]
+
+## Reuse and tune pre-trained model
+
+1. Create the `ReuseAndTuneSentimentModel()` method, just before the `IntermediateFeatures` utility class, using the following code:
+
+    ```csharp
+    public static ITransformer ReuseAndTuneSentimentModel(MLContext mlContext)
+    {
+
+    }
+    ```
+
+    The `ReuseAndTuneSentimentModel()` method executes the following tasks:
+
+    * Loads the data
+    * Extracts and transforms the data.
+    * Scores the TensorFlow model.
+    * Tunes (retrains) the model.
+    * Returns the model.
+
+1. Create an instance of `IMDBSentiment` called `trainData` and pass it to the `Prediction Engine` by adding the following as the next lines of code in the `ReuseAndTuneSentimentModel()` method:
+
+   [!code-csharp[CreateTrainData](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#CreateTrainData)]
+
+### Load the data
+
+Data in ML.NET is represented as an [IDataView class](xref:Microsoft.ML.IDataView). `IDataView` is a flexible, efficient way of describing tabular data (numeric and text). Data can be loaded from a text file or in real time (for example, SQL database or log files) to an `IDataView` object.
```

Comment:
"Other sources" is  better than "real time" as these other sources are necessarily real time